tests/test_async.py

```diff
@@ -0,0 +1,45 @@
+import os
+import pytest
+import time
+
+from tests.util import hash
+import toolchest_client as toolchest
+
+toolchest_api_key = os.environ.get("TOOLCHEST_API_KEY")
+if toolchest_api_key:
+    toolchest.set_key(toolchest_api_key)
+
+
+@pytest.mark.integration
+def test_async_execution():
+    """
+    Tests Kraken 2 running async using a small reference database
+    """
+
+    test_dir = "test_async_execution"
+    os.makedirs(f"./{test_dir}", exist_ok=True)
+    output_dir_path = f"./{test_dir}/"
+    output_file_path = f"{output_dir_path}kraken2_output.txt"
+
+    custom_db = "s3://toolchest-fsx-databases/kraken2/k2_viral_20210517/"
+    toolchest_run = toolchest.kraken2(
+        read_one="s3://toolchest-integration-tests/synthetic_bacteroides_reads.fasta",
+        custom_database_path=custom_db,
+        output_path=output_dir_path,
+        is_async=True,
+    )
+
+    run_status = ''
+    while run_status != toolchest.Status.READY_TO_TRANSFER_TO_CLIENT:
+        time.sleep(5)
```

Comment:
Have you checked average run time for the test? 5 seconds seems non representative of actual use and even a 30 second sleep wouldn't slow down integration tests too much

Reply:
The runtime for this test is 1:15 at most :)

Not worried about the wasted cycles here

Reply:
Yeah that's short enough that 5 seconds would a representative time anyways.